applications/server/controllers/editor.py

```diff
@@ -0,0 +1,78 @@
+
+# -*- coding: utf-8 -*-
+# this file is used to test the editor
+
+
+from datetime import datetime
+
+def edit_procedure():
+    """
+    return json data to show the editor content
+    """
+    preferences={'theme':'web2py', 'editor': 'default', 'closetag': 'true', 'codefolding': 'false', 'tabwidth':'4', 'indentwithtabs':'false', 'linenumbers':'true', 'highlightline':'true'}
+    code_id = request.vars.procedure_id
+
+    #the final edition will use Team 2 API "get_procedure_data(procedure_id, stable)"  to get the data
+    #we debug with temporary table to validate the function of editor
+    data =  db(db.coding.id == code_id).select(db.coding.procedures).first().procedures
+
+    file_details = dict(
+                    editor_settings=preferences,
+                    id=code_id,
+                    data=data,
+                    )
+    plain_html = response.render('editor/edit_js.html', file_details)
+    file_details['plain_html'] = plain_html
+    return response.json(file_details)
+
+def save_procedure():
+    """
```

Comment:
Can you please write comments that include the roles of all the parameters passed here? 
As a reader, I need to understand what this is doing precisely without reading the code.